docs/07-lifecycle.md

```diff
@@ -0,0 +1,18 @@
+# App Lifecycle
+## Development
+## Submit to Marketplace
+## Provision
```

Comment:
@iomodo - would be fitting to include (or link) here to the document for provisioning/install - updated version of https://docs.google.com/document/d/1A5qnugP3Yf3Jay9IA1Iwj7jKgd8-pNeyOovpAh6qj8E/edit#heading=h.dk1m0eifigoo